packages/snap-preact/src/Snap.tsx

```diff
@@ -55,9 +55,20 @@ export type ExtendedTarget = Target & {
 	renderAfterSearch?: boolean;
 };
 
+export type SnapConfigControllerDefinition<ControllerConfig> = {
+	config: ControllerConfig;
+	targeters?: ExtendedTarget[];
+	services?: SnapControllerServices;
+	url?: UrlTranslatorConfig & {
+		initial?: InitialUrlConfig;
+	};
+	context?: ContextVariables;
+};
+
 export type SnapConfig = {
 	features?: SnapFeatures;
 	mode?: keyof typeof AppMode | AppMode;
+	pageType?: 'search' | 'category';
```

Comment:
I'm not sure this should be configurable here - the current change is around prefetching, which they could set themselves in the targeter config. I'm thinking we remove this...

In the future for Beacon 2.0 work with search vs. category, I'm thinking we could rely on explicit definition of the page type via context variables OR on the explicit setting of the search controller.